server/src/test/scala/za/co/absa/atum/server/api/database/runs/functions/CreateOrUpdateAdditionalDataSpec.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.database.runs.functions
+
+import org.junit.runner.RunWith
+import za.co.absa.atum.model.dto.{AdditionalDataSubmitDTO, PartitionDTO}
+import za.co.absa.atum.server.ConfigProviderSpec
+import za.co.absa.atum.server.api.TestTransactorProvider
+import za.co.absa.atum.server.api.database.PostgresDatabaseProvider
+import za.co.absa.fadb.exceptions.DataNotFoundException
+import za.co.absa.fadb.status.FunctionStatus
+import zio._
+import zio.test._
+import zio.test.junit.ZTestJUnitRunner
+
+@RunWith(classOf[ZTestJUnitRunner])
+class CreateOrUpdateAdditionalDataSpec extends ConfigProviderSpec {
```

Comment:
Technically, also according to @miroslavpojer , we are overtesting here. We are testing fa-db type conversion here. Not atum.

Reply:
There is quite a lot going on here with the implicit data types conversions. I think testing this is crucial. Notice we don't test different inputs, rather only one simple test case.

Reply:
I quite agree with Pavel here